### ISSUE TITLE
3 page test page in double side mode (2 paper sheet)

### ISSUE THREAD

**Issue by jmrcpn** [OpenPrinting/cups-filters#102]
Printing a test page with CUPS 2.2.12 while printer is defined with duplexer (double-side),
3 pages are printed on the two sheets printed.
I was expecting (as with cups-1.4.2) to have only 2 pages (one sheet recto-verso).
According michaelrsweet, this "bug" is cups-filters one.

Note:
- test page in none duplexer mode (single side) is OK.
- Problem is occurring with different printer brand  (HP, lexmark)
- Printing a standard file in duplexer mode (seems) to be quite correct 

Small problem with the test-page routine in duplexer mode?

**Comment by jmrcpn** [comment]
Bug still present with cups-filters 1.22.3 (cups version 2.2.10)

**Comment by tillkamppeter** [comment]
@sahilarora535, could you have a look into this? It seems to be a problem in your new bannertopdf filter.

Reproducer:
```
cat /usr/share/cups/data/testprint | PPD=Printer.ppd /usr/lib/cups/filter/bannertopdf 1 1 1 1 "Duplex=DuplexNoTumble" > out.pdf
```
with an arbitrary PPD file (will probably need "Duplex" option), or even without PPD:
```
cat /usr/share/cups/data/testprint | /usr/lib/cups/filter/bannertopdf 1 1 1 1 "Duplex=DuplexNoTumble" > out.pdf
```
If duplex printing is set, like here there are 3 instead of the expected 2 output pages. With Duplex not set, there is as expected one output page.
By the way, in all of the cases the first page has the number 2 and not 1.

**Comment by tillkamppeter** [comment]
@sahilarora535, I have already found and fixed the bug. I will commit the fix soon.

**Comment by jmrcpn** [comment]
Good, let us know when commit is done
such I can confirm it is ok.

**Comment by tillkamppeter** [comment]
Fixed in commit 8b51968106.
Thank you for the bug report.

**Comment by paulmenzel** [comment]
This is Debian bug report [cups-filters: Printing a test page prints 3 pages on duplex printer][1].

[1]: https://bugs.debian.org/922409
